Updated TOC as a first step towards standard docs.microsoft.com struc…

v-stgr opened the pull request:
…ture. No articles under the TOC have been updated yet.
This TOC may need to go through a few iterations for this PR to get the formatting right and to integrate in reviewer feedback, so please feel free to comment on this PR.
After this TOC has been integrated into the published quantum docs. The next phase will be to update some of the articles under the updated TOC including the initial landing page, overview page, etc. Typically at one page per PR.
Thanks.

opbld32 commented:
Docs Build status updates of commit _[830e397](https://github.com/v-stgr/quantum-docs-pr/commits/830e397ac5b3209611450b4fa8eb8817aa16f344)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/TOC.yml](https://github.com/v-stgr/quantum-docs-pr/blob/branch3/articles/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/quantum/techniques/working-with-qubits?view=qsharp-preview&branch=pr-en-us-505) (qsharp-preview) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C29%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201908292149291348-505%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=ng1gxevripfOYJkQ5sUdAEGCTsCLrOGYUPR9R571TA0%3D&st=2019-08-29T21%3A45%3A56Z&se=2019-09-29T21%3A50%3A56Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

cgranade commented on articles/TOC.yml:
```diff
@@ -1,33 +1,54 @@
+  items:
+  - name: Glossary
+    href: Glossary.md
+  - name: Q# libraries
```
At the moment, the libraries section isn't really written as a reference. Does that cause any problems here?

v-stgr replied:
I think it looks close enough for a first draft of the TOC reorg.  Anyone else?

cgranade commented on articles/TOC.yml:
```diff
@@ -1,33 +1,54 @@
+    href: install-guide/python.md
+- name: Concepts
+  items:
+  - name: Quantum computing concepts
```
The "quantum computing concepts" section is disconnected enough from Q# and the Quantum Development Kit that I worry a little bit about it being the _only_ item listed under "Concepts." Some of the content elsewhere about the Q# language and libraries might could come forward to here?

v-stgr replied:
I'm not sure. Perhaps other team members will chime in with their thoughts after the holiday weekend.  Thanks for pointing this out.

geduardo replied:
I agree with @cgranade, probably the only item listed under concepts would be "quantum computing concepts". I didn't find anything in Q# Language that could have its subsection there. I think some parts of the "Q# standard libraries" could fit in "quantum computing concepts". 
As an idea maybe we could create a section about "Computer programming concepts" with general concepts about programming languages. For example, I come from a pure physics background and it's not rare that I need to look up in Google some terminology of the documentation related to CS concepts. I think a short section with the main things (what's a compiler, namespaces, etc) in the context of Q# could be helpful. But I don't know to what extent it is worth it.

v-stgr replied:
I looked at the Q# libraries and Q# programming concepts sections again. I think they are conceptual enough to move over to the Concepts section, and modify as necessary in subsequent doc revisions. Meanwhile, the C# library reference and C# components reference should be moved up from the bottom of the TOC to the Reference section.

cgranade replied:
Makes sense, that sounds like a good path forward!

cgranade commented on articles/TOC.yml:
```diff
@@ -1,33 +1,54 @@
+    href: libraries/toc.yml
+  - name: Q# programming language
+    topicHref: language/index.md
+    href: language/toc.yml
```
The Q#, .NET, and Python APIs are all also reference items, should they be linked to from here?

v-stgr replied:
The Q# Library Reference and C# Components Reference are currently linked from the TOC.yml file as they exist outside of docs.microsoft.com. I also would like to move them under References and need to discuss how to move them more with Carolyn and Natalie.  I'm not sure how they are referenced for the TOC at the moment.  Good call in noticing this.  I didn't see any Python specific library docs for the Quantum SDK. Is there a link you can share?  Thanks.

cgranade commented on articles/TOC.yml:
```diff
@@ -1,33 +1,54 @@
+  items:
+  - name: Quantum blog
+    href: https://cloudblogs.microsoft.com/quantum/
+  - name: Q# blog
```
```suggestion
  - name: Q# development blog
```

v-stgr replied:
I will change this.

cgranade commented on articles/TOC.yml:
```diff
@@ -1,33 +1,54 @@
+    topicHref: contributing/index.md
+- name: Quickstarts
+  items:
+  - name: C#
```
I like separating out quickstarts like this, but at the moment it feels like a mix of install instructions (C#, Jupyter / IQ#, and Python) and of quickstarts that help users write their first Q# program without reference to the choice of development environment or host language. For example, I don't see "Create a Bell state in Q#" as an alternative task to "Python," but as something I might want to do once I have Python set up to work with Q#.

v-stgr replied:
Hi Chris - Natalie asked that I submit an initial rework of the TOC without restructuring any articles within the existing site.  So that is why the "Getting started" material is under quickstarts, even though it is mostly prerequisite material.  I have reworked the Bell State in Q# quickstart offline to have a prerequisite section that contains the necessary steps that are currently in Getting started.  Typically Azure programming articles have a prerequisite section in case a reader navigates to it directly from a Google search, etc.  So essentially what you see is an interim step, not the final vision for the TOC.

cgranade replied:
Makes sense. We may want to re-examine the concepts in the Bell state quickstart as well, then, to be clear about the learning goals of that page. At the moment, we've treated Bell state preparation as a stand-in for a simple task one could perform as a part of a larger Q# application; it's nice from the standpoint that also demonstrates an example of entanglement, but we have definitely seen issues where people have confused "Bell state preparation" with "Bell tests" (aka CHSH games).

Promoting Bell state preparation to the title of the quickstart puts the specifics of the example ahead of the more general learning outcomes we had previously (see below), such that I agree with this being an interim step that will need some more iteration.

![image](https://user-images.githubusercontent.com/31516/64291088-3531c980-cf1c-11e9-828f-209374a58c21.png)

geduardo commented:
"Quantum development techniques" and "Q# programming language" are being merged into a bigger single section with the tentative title "Q# manual" (#486). Maybe you should know this for the organization of the TOC.
And what do you think about replacing the "How-to guides" section with a bigger section "Learning resources" (or something similar) in which the "How-to guides" are included as well as other tutorials and learning resources (more info about my idea of this section in #487)?

opbld31 commented:
Docs Build status updates of commit _[13ea8ee](https://github.com/v-stgr/quantum-docs-pr/commits/13ea8ee4a037de78ab49675353720efcaa1dcc5a)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/TOC.yml](https://github.com/v-stgr/quantum-docs-pr/blob/branch3/articles/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/quantum/techniques/working-with-qubits?view=qsharp-preview&branch=pr-en-us-505) (qsharp-preview) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C3%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201909031729017048-505%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Qe%2BdJwM609bJwoCYJE60CY4owzn7EE%2BKcliIvEW3ijw%3D&st=2019-09-03T17%3A25%3A26Z&se=2019-10-04T17%3A30%3A26Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

opbld34 commented:
Docs Build status updates of commit _[3900ceb](https://github.com/v-stgr/quantum-docs-pr/commits/3900ceb4d0778908f3c40ba884ed1a581618212e)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/TOC.yml](https://github.com/v-stgr/quantum-docs-pr/blob/branch3/articles/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/quantum/techniques/working-with-qubits?view=qsharp-preview&branch=pr-en-us-505) (qsharp-preview) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C3%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201909032054300769-505%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=i1m3JMKDHYwUNbj3WQNK5%2FHYX4MHwIKo0hBolZlnxD0%3D&st=2019-09-03T20%3A51%3A06Z&se=2019-10-04T20%3A56%3A06Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

opbld31 commented:
Docs Build status updates of commit _[aa42f50](https://github.com/v-stgr/quantum-docs-pr/commits/aa42f50d5c72a860e8e8023e39a416fb3d103d14)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/TOC.yml](https://github.com/v-stgr/quantum-docs-pr/blob/branch3/articles/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/quantum/techniques/working-with-qubits?view=qsharp-preview&branch=pr-en-us-505) (qsharp-preview) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C3%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201909032105043978-505%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=NvYdnc5mHRHWjMj5LKS8XkvxbrkdN8EOek4Y6auGw1o%3D&st=2019-09-03T21%3A01%3A25Z&se=2019-10-04T21%3A06%3A25Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

v-stgr commented:
Based on your feedback, I changed the Q# blog title, moved Q# libraries and Q# programming language to Concepts, and added Q# library and C# components to reference sections.

The Q# library reference and C# components reference at the bottom of the TOC have to be updated in a separate file as part of the build system.

Please have a look and let me know how it looks.  Is this acceptable for a first draft TOC change?

v-stgr commented:
@bromeg

opbld31 commented:
Docs Build status updates of commit _[d63fe72](https://github.com/v-stgr/quantum-docs-pr/commits/d63fe726123acc9f61df125df1ffff8e0e639be8)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[articles/TOC.yml](https://github.com/v-stgr/quantum-docs-pr/blob/branch3/articles/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/quantum/for-more-info?view=qsharp-preview&branch=pr-en-us-505) (qsharp-preview) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C9%5C6%5C509346e3-2203-7eb3-98f8-84fb7fa5518f%5CPullRequest%5C201909061957084526-505%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=G77kiMre9ocQ7efhSPuueUNnI64OYSuJ8O5Ip2FgRBU%3D&st=2019-09-06T19%3A53%3A37Z&se=2019-10-07T19%3A58%3A37Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

bromeg commented:
Looks good to me--I agree with Chris' point about needing to rework the Quickstart section so that the different paths/intent of each article is clear. For some reason in the preview I see the library and component references appearing twice--once in "References" and once after the rest of the nav pane. I couldn't figure out where this was happening based on the TOC.yml file, but we definitely don't want to duplicate those sections.

v-stgr commented:
Thanks, Megan. The additional components references at the bottom are defined elsewhere, but I'm not sure where otherwise I would have taken them out of the preview (Carolyn and I dicussed these at one point). Perhaps Natalie or someone else knows where these are defined in the TOC?

natke commented:
This is a great start to refactoring the TOC and gives us a taste of where we want to head. We need to flesh out the getting started experience, and add some tutorials before we publish this TOC change, so that we can have maximum impact. I will close for now. We can revisit these changes in a couple of weeks when we have re-authored the getting started experience and developed one or two tutorials.
